bucket_map/src/bucket_storage.rs

```diff
@@ -126,16 +131,21 @@ impl BucketStorage {
         )
     }
 
-    pub fn uid(&self, ix: u64) -> Uid {
+    pub fn uid(&self, ix: u64) -> Option<Uid> {
         assert!(ix < self.capacity(), "bad index size");
         let ix = (ix * self.cell_size) as usize;
         let hdr_slice: &[u8] = &self.mmap[ix..ix + std::mem::size_of::<Header>()];
         unsafe {
             let hdr = hdr_slice.as_ptr() as *const Header;
-            return hdr.as_ref().unwrap().uid();
+            hdr.as_ref().unwrap().uid()
         }
     }
 
+    /// caller knows id is not empty
+    pub fn get_known_uid(&self, ix: u64) -> Uid {
+        self.uid(ix).unwrap()
+    }
```

Comment:
Do you anticipate this fn getting called in more places? I think I prefer seeing the `.unwrap()`. That, or maybe this could be called `uid_unchecked()` instead, since that seems to be how Rust names these kind of functions.